bash_conditionals_loops/bash_conditionals_loops.md

```diff
@@ -0,0 +1,620 @@
+
+"**If** the file is empty, **then** delete the file."
+
+Using a conditional clause like this can automate the process so that the computer checks whether the file is true rather than you having to do it!
+
+That said, the Bash language does not make it easy to write conditional statements. Over the next few pages you will learn
+
+* how conditionals work in general (not specific to the Bash language)
+* how to recognize conditionals in Bash
+* how to make small changes to existing code.
+
+If you find that you need to use conditionals a lot in Bash, you will probably settle on a favorite cheat sheet, of which there are many. (links here and in additional resources)
+
+### Statements
+
+A statement (or mathematical statement) is something that can be evaluated as either true or false. Here are some examples:
+
+- 5 = 7
+- 100 is greater than 50
+- The string "blue" appears in this file's name
+- This file is empty
+- The contents of these two files are identical
+
+The first statement is false, the second one is true, and the last three can be determined to be true or false as soon as "this file" or "these two files" are identified.
+
+Opinions, questions, and instructions are among the many things that we can say that do not fall into the category of mathematical statements.
+
+<div class = "warning">
+
+**True / False in Bash**
+
+Bash is an older language and can be used to on all sorts of computers with different operating systems to interface with file and directories. This versatility means that Bash shows up in a lot of places. Unfortunately that does not mean that Bash is easy to understand. In fact the opposite is true.
+
+One of Bash's worst properties is that, unlike most programming languages which think of "true" statements as `1` and "false" statements as `0`, in Bash a "true" statement is assigned the numerical value `0` (except when evaluating mathematical expressions inside of double parentheses).
+
+</div>
+
+### If / Then Structure
+
+The basic structure of an if / then statement in Bash is:
+
+```
+if STATEMENT
+then RUN THIS COMMAND
+fi
+```
+The three lines together make up the conditional statement.
+
+* `if` tells the computer to check whether the statement following it is true or false
+* `then` tells the computer what to do if the statement is true
+* `fi` tells the computer that the conditional statement is completed.
+
+Let's see a working example:
+
+```
+if [ 5 -eq 05 ]
+then echo "integer equality"
+fi
+```
+
+<div class = "important">
+
+**Bash's `test` utility**
+
+A Bash test statement looks like this:
+
+`[ statement here ]`
+
+Notice that the statement is surrounded by square brackets with a space after the opening `[` and before the closing `]`
+
+</div>
+
+The statement `[ 5 -eq 05 ]` is checking if the numbers `5` and `05` are equal to each other. The command `echo "integer equality"` only runs if the numbers are equal. If you change the `05` to `04`, this code will not return any output.
```

Comment:
Thinking that maybe a highlight box would be useful here, to really highlight that integer equality uses `-eq` and not `=`, since that's probably not intuitive.

Reply:
Added a sentence about `=` and mentioned that it will be addressed more later in the module.